tsdb/docs/format/wal.md

```diff
@@ -85,4 +85,34 @@ and specify an interval for which samples of a series got deleted.
 │ └─────────┴───────────────────┴───────────────────┘ │
 │                        . . .                        │
 └─────────────────────────────────────────────────────┘
+<<<<<<< HEAD
+=======
 ```
+
+### Exemplars records
+
+Exemplars records encode exemplars as a list of quadruples `(series_id, timestamp, max_time, exemplar)`.
+Series reference and timestamp are encoded as deltas w.r.t the first sample.
+The first row stores the starting id and the starting timestamp.
+The first exemplar record begins at the second row.
+
+```
+┌──────────────────────────────────────────────────────────────────────────────────────────────┐
+│ type = 3 <1b>                                                                                │
+├──────────────────────────────────────────────────────────────────────────────────────────────┤
+│ ┌────────────────────┬───────────────────────────┐                                           │
+│ │ id <8b>            │ timestamp <8b>            │                                           │
+│ └────────────────────┴───────────────────────────┘                                           │
+│ ┌────────────────────┬───────────────────────────┬─────────────┬───────────────────────────┐ │
+│ │ id_delta <uvarint> │ timestamp_delta <uvarint> │ value <8b>  │ n = len(labels) <uvarint> | │
+│ ├─────────┴────────────┬───────────────────────────────────────────────────────────────────┤ │
+│ │ len(str_1) <uvarint> │ str_1 <bytes>                                                     │ │
+│ ├──────────────────────┴───────────────────────────────────────────────────────────────────┤ │
+│ │  ...                                                                                     │ │
+│ ├───────────────────────┬──────────────────────────────────────────────────────────────────┤ │
+│ │ len(str_2n) <uvarint> │ str_2n <bytes>                                                   │ │
```

Comment:
I am lost here. The above description says it's a list of `(series_id, timestamp, max_time, exemplar)`, but this seems to be something different. Can you explain?

Reply:
Maybe the `exemplar` part needs more description here. Looking at the code, makes a little more sense, but still confused a bit.